File: src/php/t4/s1.php
<?php

declare(strict_types=1);

function jump_out_of_array(array $a): int {
    $sum = 0;
    $jumps = 1;

    // If first entry in array is negative, then the first jump is out of the array
    if($a[0] < 0) {
        return $jumps;
    }

    for ($i=0; $i<= count($a); $i++) {
        $p = $a[$sum];
        $sum = $sum + $p;

        // Handle jumps out of array at the beginning of the array
        if($sum < 0) {
            return $jumps;
        }

        $jumps++;

        // Handle jumps out of array at the end of the array
        if (($sum + $p) > count($a)
        ) {
            return $jumps;
        }

        if(($i + 1) === count($a)) {
            return -1;
        }
    }
}

File: src/go/t4/s1.go
package main

import "fmt"

func JumpOutOfArray(numbers []int) int {
	sum := 0
	jumps := 0

	for index, number := range numbers {
		sum += numbers[sum]

		// Jump out of array at the start of the array
		if (index + number) < 0 {
			jumps++
			break
		}

		jumps++

		// Jump out of array at the end of the array
		if sum > len(numbers) {
			break
		}

		// End of array but do not jump out
		if (index + 1) == len(numbers) {
			jumps = -1
		}
	}

	return jumps
}

func main() {
	fmt.Println(JumpOutOfArray([]int { 2, 3, -1, 1, 6 }))
}